index.js

```diff
@@ -0,0 +1,121 @@
+import {
+  renderTopicsCard,
+  removeLoading,
+  toggleFavCardsSection,
+  renderFilterdCategories,
+  setThemeMode,
+} from "./index.view.js";
+
+const favButton = document.querySelector(".favorite-Button");
+const searchInput = document.querySelector(".search-input");
+const sortInput = document.querySelector('select[name="sort"]');
+const selectFilterElement = document.querySelector('select[name="filter"]');
+
+let data = [];
+let cards = [];
+let cardCategories = [];
+let searchTerm = "";
+let sortValue = "";
+let filterValue = "";
+let debounceSearch;
+
+let fetchTopicsList = async () => {
+  try {
+    const response = await fetch("https://tap-web-1.herokuapp.com/topics/list");
+
+    if (!response.ok) {
+      throw new Error("Something went wrong. Web topics failed to load.");
+    }
+
+    data = await response.json();
+  } catch (error) {
+    console.error("Something went wrong. Web topics failed to load.");
+  }
+};
+
+let prepareCardCategories = () => {
+  for (let i = 0; i < data.length; i++) {
+    if (!cardCategories.includes(data[i]["category"])) {
+      cardCategories.push(data[i]["category"]);
+    }
+  }
+  renderFilterdCategories(cardCategories);
+};
+
+let init = async () => {
+  setThemeMode();
+  await fetchTopicsList();
+  prepareCardCategories();
+  cards = JSON.parse(JSON.stringify(data));
+  removeLoading();
+  renderTopicsCard(cards);
+};
+
+let handleSearchInput = () => {
+  cards = data.filter((card) => {
+    return card["topic"].toLowerCase().includes(searchTerm.toLowerCase());
+  });
+};
+
+let handleSortInput = () => {
+  switch (sortValue) {
+    case "title":
+      cards.sort((a, b) => {
+        return a.topic.toLowerCase() >= b.topic.toLowerCase() ? 1 : -1;
+      });
+      break;
+    case "author":
+      cards.sort((a, b) => {
+        return a.name.toLowerCase() >= b.name.toLowerCase() ? 1 : -1;
+      });
+      break;
+    case "reset":
+      handleSearchInput();
+      break;
+  }
+};
+
+let handleFilterInput = () => {
+  cards = cards.filter((card) => {
+    return card["category"].includes(filterValue);
+  });
+};
```

Comment:
Usually, filters save their state in the URL as a search prams.  this will be much job todo here, but please do this in the react version.